refactor(docu): deployment docu (#6057)
Ocelot-Social-Community/Ocelot-Social: deployment/README.md

# Ocelot.Social Deployment And Rebranding

> The `deployment` directory is about how to change the look and feel of ocelot.social (called branding) and how to get the software running on a kubernetes cluster (deployment).

Reviewer: ```suggestion
The `deployment` directory is a module to build ocelot.social with different look and feel (Branding).
In addition it can deploy the compiled code to some Kubernetes cluster.
```

## Live demo

__Try out our deployed [development environment](https://stage.ocelot.social).__

> This environment is the vanilla ocelot without branding and with seed data.

Reviewer 2: Because this is a readme in the general deployment, where `yunite.me` and other will be placed, `vanilla ocelot` doesn't fit.

### Logins

| email | password | role |
| :--- | :--- | :--- |
| `user@example.org` | 1234 | user |
| `moderator@example.org` | 1234 | moderator |
| `admin@example.org` | 1234 | admin |

## Software requirements

- [kubectl](https://kubernetes.io/docs/tasks/tools/)
- [helm](https://helm.sh/docs/intro/install/)

Depending on your Kubernetes providers you deploy on:
| Cluster | Type | Tool |
| DigitalOcean | Cloud Hosting | [doctl](https://docs.digitalocean.com/reference/doctl/how-to/install/) |

> | Minicube | Local Deployment | [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/) |

Reviewer 2: This you design as a table, but looks like this:

<img width="1106" alt="Bildschirmfoto 2023-02-28 um 11 01 58" src="https://user-images.githubusercontent.com/25344101/221820713-4dbff9ae-5fcb-4b21-a5fc-1df9e22cd54f.png">

If you're just getting started, Minikube is a tool that you can use to get your feet wet.

## Usage

The following tasks can be achieved with the tools contained in this directory:

### Start branded docker environment locally

> For the developers we offer a `docker-compose` in this directory.

Reviewer 2: Modern notation:

```suggestion
For the developers we offer a `docker compose` in this directory.
```

Or as brand:

```suggestion
For the developers we offer a Docker Compose in this directory.
```

1. Copy `.env.dist` and rename it to `.env`
2. (optional) change the `CONFIGURATION` in the `.env` value according to your desired brand. By default its `example`
3. Run `docker-compose up`

> This will start the ocelot services within docker in the selected branded version.

Reviewer 2: ```suggestion
This will start the ocelot services within Docker in the selected branded version.
```

> If you stop the docker-compose, change the value of `CONFIGURATION` in the `.env` and restart with `docker-compose up` the branded version will change.

Reviewer 2: ```suggestion
If you stop the `docker compose`, change the value of `CONFIGURATION` in the `.env` and restart with `docker compose up` the branded version will change.
```

> Note: this might require `export DOCKER_BUILDKIT=0` to be set in order to have docker-compose build correctly.

Reviewer 2: ```suggestion
Note: this might require `export DOCKER_BUILDKIT=0` to be set in order to have `docker compose` build correctly.
```

Or as brand:

```suggestion
Note: this might require `export DOCKER_BUILDKIT=0` to be set in order to have Docker Compose build correctly.
```

### Install a cluster

> TODO

Reviewer 2: May add that the example folder has to be copied and renamed?
